packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Wysiwyg/Editor.tsx

```diff
@@ -60,12 +76,55 @@ const Editor = React.forwardRef<EditorApi, EditorProps>(
         inputStyle: 'contenteditable',
       });
 
-      // @ts-expect-error – doesn't think command exists?
+      // Register custom commands for markdown formatting
       CodeMirror.commands.newlineAndIndentContinueMarkdownList =
         newlineAndIndentContinueMarkdownList;
-      editorRef.current.on('change', (doc) => {
+
+      CodeMirror.commands.bold = () => markdownHandler(editorRef, 'Bold');
+      CodeMirror.commands.italic = () => markdownHandler(editorRef, 'Italic');
+      CodeMirror.commands.underline = () => markdownHandler(editorRef, 'Underline');
+      CodeMirror.commands.link = () => markdownHandler(editorRef, 'Link');
+
+      const changeHandler = (doc: CodeMirror.Editor) => {
         onChangeRef.current(name, doc.getValue());
-      });
+      };
+
+      editorRef.current.on('change', changeHandler);
+
+      // Handle paste event to detect images
+      const handlePaste = (_cm: CodeMirror.Editor, event: ClipboardEvent) => {
+        const items = event.clipboardData?.items;
+        if (!items) return;
+
+        // Collect all image files
+        const imageFiles: File[] = [];
+        for (let i = 0; i < items.length; i++) {
+          const item = items[i];
+          if (item.type.indexOf('image') !== -1) {
+            const file = item.getAsFile();
+            if (file) {
+              imageFiles.push(file);
+            }
+          }
+        }
```

Comment:
to match our code style better, could you use a combination of Array .filter and .map for this?